tips/TIP-0018/tip-0018.md

```diff
@@ -0,0 +1,3767 @@
+                        </td>
+                    </tr>
+                    <tr>
+                        <td>Address</td>
+                        <td>ByteArray[20]</td>
+                        <td>The raw bytes of the <i>Alias ID</i> which is the BLAKE2b-160 hash of the outputID that created it.</td>
+                    </tr>
+                </table>
+            </details>
+            <details>
+                <summary>NFT Address</summary>
+                <table>
+                    <tr>
+                        <td><b>Name</b></td>
+                        <td><b>Type</b></td>
+                        <td><b>Description</b></td>
+                    </tr>
+                    <tr>
+                        <td>Address Type</td>
+                        <td>uint8</td>
+                        <td>
+                            Set to <strong>value 16</strong> to denote an <i>NFT Address</i>.
+                        </td>
+                    </tr>
+                    <tr>
+                        <td>Address</td>
+                        <td>ByteArray[20]</td>
+                        <td>The raw bytes of the <i>NFT ID</i> which is the BLAKE2b-160 hash of the outputID that created it.</td>
+                    </tr>
+                </table>
+            </details>
+        </td>
+    </tr>
+</table>
+
+| :information_source:  Good to know about address format |
+|----------------------------------------------------------|
+
+The <i>Address Type</i> byte of a raw address has an effect on the starting character of the bech32 encoded address,
+which is the recommended address format for user facing applications.
+
+A usual bech32 encoded mainnet address starts with `iota1`, and continues with the bech32 encoded bytes of the address.
+By choosing <i>Address Type</i> as a multiple of 8 for different address types, the first character after the `1`
+separator in the bech32 address will always be different.
+
+| Address | Type Byte as `uint8` | Bech32 Encoded |
+| ------- | --------- | -------------- |
+| Ed25519 | 0 | iota1**q**... |
+| Alias | 8 | iota1**p**... |
+| NFT | 16 | iota1**z**... |
+
+A user can identify by looking at the address whether it is a signature backed address, a smart contract chain account
+or an NFT address.
+
+##### Dust Deposit Return Unlock Condition
+
+This unlock condition is employed to achieve conditional sending. An output that has <i>Dust Deposit Unlock Condition</i>
+specified can only be consumed in a transaction that deposits `Return Amount` IOTA coins into `Return Address`. When
+several of such outputs are consumed, their return amounts per `Return Addresses` are summed up and the output side
+must deposit this total sum per `Return Address`.
+
+###### Additional syntactic transaction validation rule:
+
+- `Return Amount` must be ≥ than `Minimum Dust Deposit` and must not be `0`.
+- `Return Amount` in a <i>Dust Deposit Unlock Condition</i> must be ≤ than the required dust deposit of the output.
```

Comment:
What's the reason for this limitation?

Reply:
> `Return Amount` must be ≥ than `Minimum Dust Deposit` and must not be `0`.
Otherwise the recipient can't create the output that returns the dust deposit, as it would violate dust rules. It can not  be 0 because then the unlock condition doesn't have any affect, so why let it be included anyway?

> `Return Amount` in a <i>Dust Deposit Unlock Condition</i> must be ≤ than the required dust deposit of the output.
Ottherwise you would be able to await more iotas than you actually used for covering the dust output, which means you are essentially creating a swap offfer. You can read more about the problems related to this in #42 .

Reply:
> Why would you lock more of your funds (Return Amount) into the output than you absolutely have to? What does it achieve?

Because otherwise you have to split your funds to multiple outputs if you don't already have exactly the `Return Amount`+`100i` amount in one output, and if you don't have > 2*deposit amount IOTAs, then you can't do it at all, because the remaining IOTAs wouldn't be enough to cover the byte cost

Reply:
> Because otherwise you have to split your funds to multiple outputs if you don't already have exactly the Return Amount+100i amount in one output

Yes, it is unlikely that you have exactly `Return Amount`+`100i` in one input, so you'll have to split something up anyways. Splitting up but still not locking it into the DDR output is stil ldesried, no?

> and if you don't have > 2*deposit amount IOTAs, then you can't do it at all, because the remaining IOTAs wouldn't be enough to cover the byte cost

Yes, this is true, however unlikely that most users will have such low balance <1Mi. If we want to cover this case, the rule would be `0` < `Amount` - `Return Amount` =< `Minimum Dust Deposit`